raiden/network/transport/matrix/transport.py

```diff
@@ -1012,12 +1015,48 @@ def _get_room_for_address(
             self.log.error('No valid peer found', peer_address=address_hex)
             return None
 
-        self._address_to_userids[address].update({user.user_id for user in peers})
-
         if self._private_rooms:
             room = self._get_private_room(invitees=peers)
         else:
             room = self._get_public_room(room_name, invitees=peers)
+
+        peer_ids = self._address_to_userids[address]
+        member_ids = {member.user_id for member in room.get_joined_members()}
+        room_is_empty = not bool(peer_ids & member_ids)
+        if room_is_empty:
+            last_ex: Optional[Exception] = False
+            retry_interval = 0.1
```

Comment:
It seems this `retry_interval` + `for range(JOIN_RETRIES)`  pattern is a trend and seems useful to properly waiting for the async tasks that depends on matrix server's reactions. Would it be worth it to put that in a separate function, possibly a generator? This would ease e.g. receiving an `Event` (`_stop_event`) as optional argument and `.wait`ing on it instead of `sleep`ing and bailing out early if it's set during the lifetime of the iteration, so these loops doesn't prevent the `Runnable` from completing once we're stopping or any exception happened elsewhere

Reply:
So if I understood @ulope correctly here, he opted for using the `JOIN_RETRIES` as more explicit. Otherwise I could also use the `_http_retry_delay` iterator that we already have

Reply:
Not necessarily always, but for those cases I think it's nice and easy and doesn't add much code. A generator adds a level of indirection that isn't really needed IMO.